src-input/duk_bi_object.c

```diff
@@ -705,6 +760,11 @@ DUK_INTERNAL duk_ret_t duk_bi_object_constructor_prevent_extensions(duk_context
 
 	magic = duk_get_current_magic(ctx);
 
+	/* ES6 Section 19.1.2.15, step 1 */
+	if (magic == 0 && !duk_is_object(ctx, 0)) {
+		goto done;  /* silent success */
+	}
+
 	if (duk_check_type_mask(ctx, 0, DUK_TYPE_MASK_LIGHTFUNC | DUK_TYPE_MASK_BUFFER)) {
```

Comment:
The above duk_is_object() test can be implemented here, by choosing the type mask based on the magic value. For readability the type mask can be computed to a variable first.